src/core/tests/main.cpp

```diff
@@ -71,7 +71,9 @@ GTEST_API_ int main(int argc, char **argv)
 
     if (g_test_ret != 0)
     {
+#ifndef ENABLE_GCOV
```

Comment:
if not use dsn_exit, it is possible the process will crash. please make sure this is still what gcov wants.